superduperdb/backends/base/data_backend.py

```diff
@@ -130,3 +138,58 @@ def auto_create_table_schema(self, db, table_name, documents):
         :param table_name: The table name
         :param documents: The documents
         """
+
+
+class DataBackendProxy:
+    """
+    Proxy class to DataBackend which acts as middleware for performing fallbacks.
+
+    :param backend: Instance of `BaseDataBackend`.
+    """
+
+    def __init__(self, backend):
+        self._backend = backend
+
+    @property
+    def datalayer(self):
+        """Return the datalayer."""
+        return self._backend._datalayer
+
+    @datalayer.setter
+    def datalayer(self, value):
+        """Set the datalayer.
+
+        :param value: The datalayer.
+        """
+        self._backend._datalayer = value
+
+    @property
+    def type(self):
+        """Instance of databackend."""
+        return self._backend
+
+    def _try_execute(self, attr):
+        @functools.wraps(attr)
+        def wrapper(*args, **kwargs):
+            try:
+                return attr(*args, **kwargs)
```

Comment:
What's this? `attr`?

Reply:
attr is the db.databackend.<attr> thing.